src/Stateless/TriggerWithParameters.cs

```diff
@@ -28,6 +28,17 @@ public TriggerWithParameters(TTrigger underlyingTrigger, params Type[] argumentT
             /// </summary>
             public TTrigger Trigger { get { return _underlyingTrigger; } }
 
+            /// <summary>
+            /// Gets the number of parameters associated with the trigger
+            /// </summary>
+            internal virtual int Arity => this.GetType()
```

Comment:
If this was defined `... Arity { get; } = ...` then initialization would occur only once, rather than every time the property is accessed.